megamek/src/megamek/common/Compute.java

```diff
@@ -912,6 +912,19 @@ public static ToHitData getRangeMods(IGame game, Entity ae, int weaponId,
                 && weapon.curMode().equals("Indirect");
         boolean useExtremeRange = game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_TACOPS_RANGE);
         boolean useLOSRange = game.getOptions().booleanOption(OptionsConstants.ADVCOMBAT_TACOPS_LOS_RANGE);
+        //Naval C3 only provides full C3 range benefits to energy weapons and guided missiles
+        boolean nc3EnergyGuided = ((wtype.getAtClass() == WeaponType.CLASS_LASER)
```

Comment:
How about wtype.hasFlag(WeaponType.F_ENERGY) || wtype.hasFlag(WeaponType.F_MISSILE) instead?

Reply:
F_Energy is ok.  F_Missile is not.  This specifically doesn't apply to unguided missiles - MRMs and RLs, which also have the F_Missile flag for AMS purposes.